documentation/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,343 @@
+# Troubleshooting
+
+TorchDynamo is still in active development, and many of the reasons for graph breaks and excessive recompilation will be fixed with upcoming support for [tracing dynamic tensor shapes](https://docs.google.com/document/d/1QJB-GOnbv-9PygGlOMXwiO9K6vVNm8sNg_olixJ9koc/edit?usp=sharing), more careful choices for guards and better tuned heurstics.
+
+In the mean time, you may need to diagnose a particular issue and determine if it is easy to work around with a change to your model, or file an issue for support.
+
+We're also actively developing debug tools, profilers, and improving our errors/warnings.  Please give us feedback if you have an issue with this infra, or an idea for an improvement.
+
+## Graph Breaks
+Given a program like this,
+
+```py
+@torchdynamo.optimize(...)
+def some_fun(x):
+    ...
+
+some_fun(x)
+...
+```
+
+Torchdynamo will attempt to compile all of the torch/tensor operations within some_fun into a single FX graph, but it may fail to capture everything into one graph.
+
+Some graph break reasons are insurmountable to TorchDynamo, and can't be easily fixed.
+- calling into a C extension other than torch is invisible to torchdynamo, and could do arbitrary things without TorchDynamo being able to introduce necessary guards to ensure that the compiled program would be safe to reuse
+
+### Identifying the cause of a graph break
+
+To identify all graph breaks in a program and the associated reasons for the breaks, `torchdynamo.explain` can be used. This tool runs TorchDynamo on the supplied function and aggregates the graph breaks that are encountered. Here is an example usage:
+
+```py
+from typing import List
+import torch
+import torchdynamo
+
+def toy_example(a, b):
+    x = a / (torch.abs(a) + 1)
+    print("woo")
+    if b.sum() < 0:
+        b = b * -1
+    return x * b
+
+explained = torchdynamo.explain(toy_example, torch.randn(10), torch.randn(10))
+print(explained[0])
```

Comment:
what would `explained[1]` and the rest of `explained` contain?
worth throwing in a one-liner comment

Reply:
We can also unpack explain's return tuple into 4 args, and keep the same names as in the function...